internal/dag/gatewayapi_processor.go

```diff
@@ -1237,20 +1237,28 @@ func gatewayPathMatchCondition(match *gatewayapi_v1alpha2.HTTPPathMatch, routeAc
 
 func gatewayHeaderMatchConditions(matches []gatewayapi_v1alpha2.HTTPHeaderMatch) ([]HeaderMatchCondition, error) {
 	var headerMatchConditions []HeaderMatchCondition
+	seenNames := sets.String{}
 
 	for _, match := range matches {
-		// HeaderMatchTypeExact is the default if not defined in the object.
-		headerMatchType := HeaderMatchTypeExact
-		if match.Type != nil {
-			switch *match.Type {
-			case gatewayapi_v1alpha2.HeaderMatchExact:
-				headerMatchType = HeaderMatchTypeExact
-			default:
-				return nil, fmt.Errorf("HTTPRoute.Spec.Rules.Matches.Headers: Only Exact match type is supported")
-			}
+		// "Exact" is the default if not defined in the object, and
+		// the only supported match type.
+		if match.Type != nil && *match.Type != gatewayapi_v1alpha2.HeaderMatchExact {
+			return nil, fmt.Errorf("HTTPRoute.Spec.Rules.Matches.Headers: Only Exact match type is supported")
+		}
+
+		// If multiple match conditions are found for the same header name (case-insensitive),
+		// use the first one and ignore subsequent ones.
+		upperName := strings.ToUpper(string(match.Name))
+		if seenNames.Has(upperName) {
+			continue
```

Comment:
a shame maybe that the spec doesn't have a status warning/condition/reason etc. to set for this sort of thing, but could be added if people start getting confused about how things work

Reply:
Yeah, that could be added -- I raised an issue upstream about the query param thing and there's been some discussion there on adding validation for these things to the webhook, so adding a Condition could be another possible outcome there, I'll keep an eye on it.